lib/cretonne/src/ir/frontend.rs

```diff
@@ -0,0 +1,764 @@
+    }
+
+    /// Returns the value corresponding to the `i`-th argument of the function as defined by
+    /// the function signature. Panics if `i` is out of bounds or if called before the first call
+    /// to `switch_to_block`.
+    pub fn arg_value(&self, i: usize) -> Value {
+        debug_assert!(!self.pristine, "you have to call switch_to_block first.");
+        self.builder.function_args_values[i]
+    }
+
+    /// Creates a jump table in the function, to be used by `br_table` instructions.
+    pub fn create_jump_table(&mut self) -> JumpTable {
+        self.func.jump_tables.push(JumpTableData::new())
+    }
+
+    /// Inserts an entry in a previously declared jump table.
+    pub fn insert_jump_table_entry(&mut self, jt: JumpTable, index: usize, ebb: Ebb) {
+        self.func.jump_tables[jt].set_entry(index, ebb);
+    }
+
+    /// Creates a stack slot in the function, to be used by `stack_load`, `stack_store` and
+    /// `stack_addr` instructions.
+    pub fn create_stack_slot(&mut self, data: StackSlotData) -> StackSlot {
+        self.func.stack_slots.push(data)
+    }
+
+    /// Adds a signature which can later be used to declare an external function import.
+    pub fn import_signature(&mut self, signature: Signature) -> SigRef {
+        self.func.dfg.signatures.push(signature)
+    }
+
+    /// Declare an external function import.
+    pub fn import_function(&mut self, data: ExtFuncData) -> FuncRef {
+        self.func.dfg.ext_funcs.push(data)
+    }
+
+
+
+    /// Returns an object with the [`InstBuilder`](../trait.InstBuilder.html) trait that allows to
+    /// conveniently append an instruction to the current `Ebb` being built.
+    pub fn ins<'short>(&'short mut self) -> FuncInstBuilder<'short, 'a, Variable> {
+        let ebb = self.position.ebb;
+        FuncInstBuilder::new(self, ebb)
+    }
+}
+
+/// All the functions documented in the previous block are write-only and help you build a valid
+/// Cretonne IL functions via multiple debug asserts. However, you might need to improve the
+/// performance of your translation perform more complex transformations to your Cretonne IL
+/// function. The functions below help you inspect the function you're creating and modify it
+/// in ways that can be unsafe if used incorrectly.
+impl<'a, Variable> FunctionBuilder<'a, Variable>
+    where Variable: EntityRef + Hash + Default
+{
+    /// Retrieves all the arguments for an `Ebb` currently infered from the jump instructions
+    /// inserted that target it and the SSA construction.
+    pub fn ebb_args(&self, ebb: Ebb) -> &[Value] {
+        self.func.dfg.ebb_args(ebb)
+    }
+
+    /// Creates a argument for a specific `Ebb` by appending it to the list of already existing
+    /// arguments.
+    ///
+    /// **Note:** this function has to be called at the creation of the `Ebb` before adding
+    /// instructions to it, otherwise this could interfere with SSA construction.
```

Comment:
I can't do that because the function has a arbitrary `ebb` argument whereas `self.position.pristine` concerns only the current `Ebb`. To enforce this assert I would need to store the `pristine` parameter inside `EbbData`.